docs/ds/dsu.md

```diff
@@ -1,150 +1,156 @@
 author: HeRaNO, JuicyMio, Xeonacid, sailordiary, ouuan
 
-并查集是一种树形的数据结构，顾名思义，它用于处理一些不交集的 **合并** 及 **查询** 问题。
-它支持两种操作：
+Disjoint set is a tree-shaped data structure. As the name suggests, it is used to deal with some disjoint **merge** and **query** problems.
+It supports two operations:
 
-- 查找（Find）：确定某个元素处于哪个子集；
+-   Find: determine which subset an element is in.
 
-- 合并（Union）：将两个子集合并成一个集合。
+-   Union: merge two subsets into one set.
 
 !!! warning
-    并查集不支持集合的分离，但是并查集在经过修改后可以支持集合中单个元素的删除操作（详见 UVA11987 Almost Union-Find）。使用动态开点线段树还可以实现可持久化并查集。
+    The disjoint set does not support the set separation. However, it supports the deletion of a single element in the set after modification (see [UVA11987 Almost Union-Find](https://onlinejudge.org/index.php?option=com_onlinejudge&Itemid=8&page=show_problem&problem=3138) for details). Dynamically opening nodes on segment tee can also realize persistent disjoint set.
 
-## 初始化
+## Initialization
 
 ```cpp
-void makeSet(int size) {
-  for (int i = 0; i < size; i++) fa[i] = i;  // i就在它本身的集合里
+void makeSet(inßt size) {
+  for (int i = 0; i < size; i++) fa[i] = i;  // i is in its own set
   return;
 }
 ```
 
-## 查找
+## Find
 
-通俗地讲一个故事：几个家族进行宴会，但是家族普遍长寿，所以人数众多。由于长时间的分离以及年龄的增长，这些人逐渐忘掉了自己的亲人，只记得自己的爸爸是谁了，而最长者（称为「祖先」）的父亲已经去世，他只知道自己是祖先。为了确定自己是哪个家族，他们想出了一个办法，只要问自己的爸爸是不是祖先，一层一层的向上问，直到问到祖先。如果要判断两人是否在同一家族，只要看两人的祖先是不是同一人就可以了。
+First let's hear a story: Several families hold banquets and there are too many people because they usually live a long life. Due to long time separation and aging, these people gradually forget their relatives and only remember who their father is. The father of the oldest person (called "ancestor") has passed away, and he only knows that he is an ancestor. . In order to determine which family they belonged to, they came up with a way, as long as they asked their father if they were ancestors, and asked them generation by generation until they asked their ancestors. If you want to know whether two people are in the same family, you only need to see whether the ancestors of the two are the same.
```

Comment:
```suggestion
First let's hear a story: Several families hold banquets and there are too many people because they usually live a long life. Due to long time separation and aging, these people gradually forget their relatives and only remember who their father is. The father of the oldest person (called "ancestor") has passed away, and he only knows that he is an ancestor. In order to determine which family they belonged to, they came up with a way, as long as they asked their father if they were ancestors, and asked them generation by generation until they asked their ancestors. If you want to know whether two people are in the same family, you only need to see whether the ancestors of the two are the same.
```
讲故事可以用过去时吧 🤔 
diff 是一个多余的句号